var/spack/repos/builtin/packages/intel-oneapi-mkl/package.py

```diff
@@ -200,10 +202,15 @@ def _find_mkl_libs(self, shared):
         if self.spec.satisfies("+cluster"):
             libs.extend([self._xlp64_lib("libmkl_scalapack"), "libmkl_cdft_core"])
 
-        if self.spec.satisfies("%oneapi") or self.spec.satisfies("%intel"):
-            libs.append(self._xlp64_lib("libmkl_intel"))
-        else:
+        # Explicit variant for compatibility with gfortran, otherwise
+        # support intel fortran. Be aware that some dependencies may
+        # be using this logic and other dependencies might be using
+        # cmake for the library list and they have to be consistent.
+        # https://github.com/spack/spack/pull/43673 for discussion
+        if self.spec.satisfies("+gfortran"):
```

Comment:
...and check the dependency like this:
```suggestion
        if self.spec.satisfies("^[virtuals=fortran-rt] gcc-runtime"):
```
Note that you might need to add a dependency on `fortran` to other packages too, for the constraint to properly take effect cross-packages.

Reply:
If the other package does not have the `depends` for  fortran, could I add `^[virtuals=fortran-rt]` to the spec on the command line?

Reply:
As long as at least one package has it, you can. But then other packages won't have an edge needing a `fortran-rt` (fortran runtime) along some edges.

Reply:
I am still confused about how the fortran-rt can be used here. I need to know if there is a package that intends to call mkl fortran interfaces, and whether that package is `%gcc` or `%oneapi` toolchain.

A package can declare that it depends on mkl and depends on fortran, but cannot declare that it depends on mkl fortran. If there are 2 packages that depend on mkl and fortran, and one package is compiled with `%gcc` and the other with `%oneapi`, then I don't know which fortran to be compatible with.

The `gfortran` variant lets the user explicitly declare which fortran interface to provide.

Reply:
After merge, will follow up with @alalazo on more precise modeling of oneapi runtime constraints.